shell/components/form/Members/ClusterPermissionsEditor.vue

```diff
@@ -175,7 +175,20 @@ export default {
         opt:  { url: `/v3/principals/${ principalId }` }
       }, { root: true });
     },
+    customPermissionsUpdate() {

```

Comment:
I think adding `locked` and `tooltip` properties to  `this.customPermissions` elements is not necessary here, since we are returning `customPermissionsUpdate` as a computed variable.
Suggested solution:
```
customPermissionsUpdate() {
  return this.customPermissions.reduce((acc, customPermissionsItem) => {
    const lockedExist = this.roleTemplates.find(roleTemplateItem => roleTemplateItem.displayName === customPermissionsItem.label);

    if (lockedExist.locked) {
      customPermissionsItem['locked'] = true;
      customPermissionsItem['tooltip'] = 'This role is locked.';
    }

    return [...acc, customPermissionsItem];
  }, []);
```